modules/firehose/driver_plan.go

```
@@ -159,3 +168,60 @@ func (fd *firehoseDriver) planReset(exr module.ExpandedResource, act module.Acti
 	}
 	return &exr.Resource, nil
 }
+
+func (fd *firehoseDriver) planReset(exr module.ExpandedResource, act module.ActionRequest) (*resource.Resource, error) {
+	resetValue, err := kafka.ParseResetParams(act.Params)
+	if err != nil {
+		return nil, err
+	}
+
+	immediately := fd.timeNow()
+
+	curConf, err := readConfig(exr.Resource, exr.Resource.Spec.Configs, fd.conf)
+	if err != nil {
+		return nil, err
+	}
+
+	curConf.ResetOffset = resetValue
+	curConf.EnvVariables[SourceKafkaConsumerAutoOffsetReset] = resetValue
+	curConf.EnvVariables[confKeyConsumerID], err = getNewConsumerGroupID(curConf.EnvVariables[confKeyConsumerID], curConf.DeploymentID)
+	if err != nil {
+		return nil, err
+	}
+
+	exr.Resource.Spec.Configs = mustJSON(curConf)
+	exr.Resource.State = resource.State{
+		Status:     resource.StatusPending,
+		Output:     exr.Resource.State.Output,
+		NextSyncAt: &immediately,
+		ModuleData: mustJSON(transientData{
+			PendingSteps: []string{
+				stepReleaseStop,   // stop the firehose
+				stepReleaseUpdate, // restart the deployment.
+			},
+		}),
+	}
+	return &exr.Resource, nil
+}
+
+func getNewConsumerGroupID(currentConsumerGroupID, deploymentID string) (string, error) {
+	const groupNumberSuffixLength = 4
+	suffix := "0000"
+
+	currentConsumerGroupSuffix := currentConsumerGroupID[len(currentConsumerGroupID)-groupNumberSuffixLength:]
+	currentConsumerGroupNumber, err := strconv.Atoi(currentConsumerGroupSuffix)
+	if err != nil {
+		return "", err
+	}
+
+	currentConsumerGroupNumber++
+	newConsumerGroupSuffix := strconv.Itoa(currentConsumerGroupNumber)
+
+	if len(newConsumerGroupSuffix) > groupNumberSuffixLength {
+		return "", errGroupNumberLimitCrossed
+	}
+
+	newConsumerGroupSuffix = suffix[:groupNumberSuffixLength-len(newConsumerGroupSuffix)] + newConsumerGroupSuffix
+
+	return fmt.Sprintf("%s-%s", deploymentID, newConsumerGroupSuffix), nil
+}
```

why do we care about the suffix length?

why not do this:

```golang
var suffixRegex = regexp.MustCompile(`^([A-Za-z0-9-]+)-([0-9]+)$`)

func getNewConsumerGroupID(curGroup) string {
    matches := suffixRegex.FindStringSubmatch(curGroup)
    prefix, sequence := matches[1], matches[2]
    
    seq, err := strconv.Atoi(sequence)
    if err != nil {
        // ignore the error. but put whatever was here back.
        prefix += sequence
    } else {
        seq++
    }
    
    return fmt.Sprintf("%s%d", prefix, seq)
}
```

---

yeah this looks good 💯 
but let's not ignore the error and propagate it back to the user? just in case someone updated it via API?
OR, we can add a suffix "-1", but there is a chance of group name confict. wdys?

---

The reason for ignoring is the initial case for example where a sequence is not added as a suffix yet.

During creation, you could simply pass the deployment ID in place of curGroup arg, and get back a valid starting point.

The same goes with update. If someone doesn't put a sequence suffix in the end, this will add one automatically

But yea, the update behaviour is something you need to take a call on.

---

hmm I get it, but just to avoid any conflicts, I think we should start with a sequence and keep increasing it.